recommendation/pytorch/convert.py

```diff
@@ -1,119 +1,60 @@
-    mlperf_log.ncf_print(key=mlperf_log.RUN_START)
-    mlperf_log.ncf_print(key=mlperf_log.INPUT_STEP_EVAL_NEG_GEN)
-
-    for user in tqdm(range(len(original_users)), desc='Users', total=len(original_users)):  # noqa: E501
-        test_item = user_to_items[user].pop()
-
-        all_ratings.remove((user, test_item))
-        all_negs = all_items - set(user_to_items[user])
-        all_negs = sorted(list(all_negs))  # determinism
-
-        test_ratings.append((user, test_item))
-        test_negs.append(list(np.random.choice(all_negs, args.negatives)))
-
-    print("Saving train and test CSV files to {}".format(args.output))
-    df_train_ratings = pd.DataFrame(list(all_ratings))
-    df_train_ratings['fake_rating'] = 1
-    df_train_ratings.to_csv(os.path.join(args.output, TRAIN_RATINGS_FILENAME),
-                            index=False, header=False, sep='\t')
-
-    mlperf_log.ncf_print(key=mlperf_log.INPUT_SIZE, value=len(df_train_ratings))
-
-    df_test_ratings = pd.DataFrame(test_ratings)
-    df_test_ratings['fake_rating'] = 1
-    df_test_ratings.to_csv(os.path.join(args.output, TEST_RATINGS_FILENAME),
-                           index=False, header=False, sep='\t')
-
-    df_test_negs = pd.DataFrame(test_negs)
-    df_test_negs.to_csv(os.path.join(args.output, TEST_NEG_FILENAME),
-                        index=False, header=False, sep='\t')
-
+    
+    train_ratings = torch.LongTensor()
+    test_chunk_size = [0] * args.user_scaling
+    for chunk in range(args.user_scaling):
+        train_ratings = torch.cat((train_ratings, 
+            torch.from_numpy(np.load(args.data + '/trainx' 
+                + str(args.user_scaling) + 'x' + str(args.item_scaling) 
+                + '_' + str(chunk) + '.npz', encoding='bytes')['arr_0'])))
+        test_ratings_chunk = torch.from_numpy(np.load(args.data + '/testx' 
+                + str(args.user_scaling) + 'x' + str(args.item_scaling) 
+                + '_' + str(chunk) + '.npz', encoding='bytes')['arr_0'])
+        test_chunk_size[chunk] = test_ratings_chunk.shape[0]
+
+    nb_maxs = torch.max(train_ratings, 0)[0]
+    nb_users = nb_maxs[0].item()+1
+    nb_items = nb_maxs[1].item()+1
+
+    sampler = NegativeSampler(train_ratings, nb_users, nb_items)
+    
+    print(datetime.now(), 'Generating negative test samples...')
+    
+    test_negatives = [torch.LongTensor()] * args.user_scaling
+    test_user_offset = 0
+    for chunk in range(args.user_scaling):
+        test_negatives[chunk] = sampler.generate_test_part(
+                args.valid_negative, 
+                test_chunk_size[chunk],
+                test_user_offset) 
+        
+        file_name = (args.data + '/test_negx' + str(args.user_scaling) + 'x'
+                + str(args.item_scaling) + '_' + str(chunk) + '.npz')
+
+        np.savez_compressed(file_name, test_negatives[chunk])
+        print(datetime.now(), 'Chunk ', chunk, '/', args.user_scaling, 'saved.')
```

Comment:
Perhaps the following?

```suggestion
        print(datetime.now(), 'Chunk ', chunk+1, '/', args.user_scaling, 'saved.')
```